apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.ValueConversion.cs

```diff
@@ -675,6 +700,19 @@ private object ConvertToClrTypeImpl(Value wireValue, System.Type targetClrType,
                             $"Invalid Type conversion from {wireValue.KindCase} to {targetClrType.FullName}");
                 }
             }
+
+            // Protobuf:
+            if (TypeCode == TypeCode.Proto
+                && wireValue.KindCase == Value.KindOneofCase.StringValue
+                && typeof(IMessage).IsAssignableFrom(targetClrType)
```

Comment:
Urgh, this is all horrible. I don't know of a better option though. Elsewhere we have a generic constraint of "implements IMessage and has a parameterless constructor" which can be used to get an instance of the message, and go from there.

Reply:
(We could potentially cache the parser though - a `ConcurrentDictionary<Type, MessageParser>`. Probably not worth doing straight away.)

Reply:
Use GetProtobufPropertyValue though?

Reply:
@jskeet Gentle ping on the parser cache part of this thread and whether we could do it with TypeRegistry.
(I think the comment got lost for your review because it was attached to changed code)

Reply:
I think it would be reasonable to add WithAddedMessages/WithAddedFiles, yes. But for now, I'd probably implement the cache (or at least consider doing so) as any change in protobuf would take a long time to roll out anyway.

Reply:
Yes, makes sense.